### PR TITLE
JAMES-3926 quota current value recomputation for sieve

### PR DESCRIPTION
**Pull request by hungphan227**
this PR is based on https://github.com/apache/james-project/pull/1707

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
```
1) [Guice/MissingImplementation]: No implementation for UploadRepository was bound.

Requested by:
1  : JMAPCurrentUploadUsageCalculator.<init>(JMAPCurrentUploadUsageCalculator.java:37)
      \_ for 1st parameter
     at RecomputeCurrentQuotasService.<init>(RecomputeCurrentQuotasService.java:180)
      \_ for 7th parameter
     at WebadminMailboxTaskSerializationModule.recomputeCurrentQuotasTask(WebadminMailboxTaskSerializationModule.java:76)
      \_ for 1st parameter
     at WebadminMailboxTaskSerializationModule.recomputeCurrentQuotasTask(WebadminMailboxTaskSerializationModule.java:76)
      \_ installed by: Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$CombinedModule -> Modules$CombinedModule -> Modules$CombinedModule -> Modules$CombinedModule -> Modules$CombinedModule -> MailboxRoutesModule -> WebadminMailboxTaskSerializationModule
2  : JMAPCurrentUploadUsageCalculator.<init>(JMAPCurrentUploadUsageCalculator.java:37)
      \_ for 1st parameter
     at RecomputeCurrentQuotasService.<init>(RecomputeCurrentQuotasService.java:180)
      \_ for 7th parameter
     at UserQuotaRoutes$RecomputeCurrentQuotasRequestToTask.<init>(UserQuotaRoutes.java:85)
      \_ for 1st parameter
     at InconsistencyQuotasSolvingRoutesModule.configure(InconsistencyQuotasSolvingRoutesModule.java:36)
      \_ installed by: Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$CombinedModule -> Modules$CombinedModule -> Modules$CombinedModule -> Modules$CombinedModule -> Modules$CombinedModule -> InconsistencyQuotasSolvingRoutesModule
```

**Review by quantranhong1999 — APPROVED** [review]
Other than the GUICE issue, it looks good to me.

**Review by quantranhong1999 — APPROVED** [review]
Other than the GUICE issue, it looks good to me.

**Comment by chibenwa** [comment]
image


Please have a look into these failures.

```
1) [Guice/MissingImplementation]: No implementation for UploadRepository was bound.

Requested by:
1  : JMAPCurrentUploadUsageCalculator.<init>(JMAPCurrentUploadUsageCalculator.java:37)
```

It's a guice architecture problem. RecomputeCurrentQuotasService should not have a hard dependency to JMAPCurrentUploadUsageCalculator. We would need to hide it with a multi-binder...

**Comment by Arsnael** [comment]
The Guice failures it's likely because this PR was built on top of https://github.com/apache/james-project/pull/1707 that had the same issue but is being fixed atm. So I would not worry too much and wait for this PR to be updated after the first one becomes valid ;)

File: mailbox/tools/quota-recompute/pom.xml [hunk 1]
```diff
@@ -46,6 +46,10 @@
             <groupId>${james.groupId}</groupId>
             <artifactId>james-server-data-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-data-jmap</artifactId>
+        </dependency>
```
Comment (chibenwa) [inline mailbox/tools/quota-recompute/pom.xml:52]: Sorry as in https://github.com/apache/james-project/pull/1707 this change is not acceptable to me.

**Review by chibenwa — CHANGES_REQUESTED** [review]

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — COMMENTED** [review]
No integration test in `server/protocols/webadmin-intergretaion-tests` ?

I just need a one tester to ensure the feature is exposed ;-)

**Comment by hungphan227** [comment]
test added

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**